proxy/forward_auth.go

```diff
@@ -11,32 +11,53 @@ import (
 	"github.com/pomerium/pomerium/internal/urlutil"
 )
 
+// registerFwdAuthHandlers returns a set of handlers that support using pomerium
+// as a "forward-auth" provider with other reverse proxies like nginx, traefik.
+//
+// see : https://www.pomerium.io/configuration/#forward-auth
 func (p *Proxy) registerFwdAuthHandlers() http.Handler {
 	r := httputil.NewRouter()
 	r.StrictSlash(true)
 	r.Use(sessions.RetrieveSession(p.sessionStore))
 
+	// NGNIX's forward-auth capabilities are split across two settings:
+	// `auth-url` and `auth-signin` which correspond to `verify` and `auth-url`
+	//
+	// NOTE: Route order matters here. As such, it's rather confusing to follow
+	// 		 so each step of the process is labeled in order of the flow
+
+	// nginx 3: save the returned session post authenticate flow
```

Comment:
What's the numbering after `nginx ` mean?

Reply:
The relative order of the steps in the flow to (hopefully?) clarify the request flow. I've adjusted the above comment. Hope that helps.